gulpfile.js

```diff
@@ -8,9 +8,17 @@ var gulp = require('gulp');
 var gutil = require('gulp-util');
 var path = require('path');
 
-gulp.task('build', ['copy-terriajs-assets', 'build-app']);
-gulp.task('release', ['copy-terriajs-assets', 'release-app', 'make-editor-schema']);
-gulp.task('watch', ['watch-terriajs-assets', 'watch-app']);
+var minNode = require('./package.json').engines.node;
+if (!require('semver').satisfies(process.version, minNode)) {
+    console.log('Terria requires Node.js ' + minNode + ' to build. Please update your version of Node.js, delete your node_modules directory' +
+        ', then run npm install and gulp again.');
+    process.exit();
+}
+
+
+gulp.task('build', ['render-datasource-templates', 'copy-terriajs-assets', 'build-app']);
```

Comment:
`render-datasource-templates` isn't needed in NM, because that's done by the separate `NationalMap-Catalog` project.